app/services/notifications/index.service.js

```diff
@@ -0,0 +1,136 @@
+  }
+}
+
+async function _getFilters(yar) {
+  const filters = yar.get('notifications-filter')
+  const filterNotificationTypes = filters?.filterNotificationTypes
+  const sentBy = filters?.sentBy
+  const sentFromDay = filters?.sentFromDay
+  const sentFromMonth = filters?.sentFromMonth
+  const sentFromYear = filters?.sentFromYear
+  const sentToDay = filters?.sentToDay
+  const sentToMonth = filters?.sentToMonth
+  const sentToYear = filters?.sentToYear
+
+  return {
+    ...(filterNotificationTypes && { filterNotificationTypes }),
+    ...(sentBy && { sentBy }),
+    ...(sentFromDay && { sentFromDay }),
+    ...(sentFromMonth && { sentFromMonth }),
+    ...(sentFromYear && { sentFromYear }),
+    ...(sentToDay && { sentToDay }),
+    ...(sentToMonth && { sentToMonth }),
+    ...(sentToYear && { sentToYear })
+  }
+}
+
+function _openFilter(filter) {
+  return (
+    (filter.filterNotificationTypes ||
+      filter.sentBy ||
+      filter.sentFromDay ||
+      filter.sentFromMonth ||
+      filter.sentFromYear ||
+      filter.sentToDay ||
+      filter.sentToMonth ||
+      filter.sentToYear) !== undefined
+  )
+}
+
+function _prepareNotifications(filterNotificationTypes) {
+  return {
+    legacyNotifications: filterNotificationTypes.includes('legacy-notifications'),
+    returnsPaperForm: filterNotificationTypes.includes('returns-paper-form'),
+    returnReminders: filterNotificationTypes.includes('returns-reminders'),
+    returnInvitation: filterNotificationTypes.includes('returns-invitation'),
+    waterAbstractionAlertResume: filterNotificationTypes.includes('water-abstraction-alert-resume'),
+    waterAbstractionAlertStop: filterNotificationTypes.includes('water-abstraction-alert-stop'),
+    waterAbstractionAlertReduce: filterNotificationTypes.includes('water-abstraction-alert-reduce'),
+    waterAbstractionAlertWarning: filterNotificationTypes.includes('water-abstraction-alert-warning')
+  }
+}
+
+function _validate(filters) {
+  const { sentFromDay, sentFromMonth, sentFromYear, sentBy, sentToDay, sentToMonth, sentToYear } = filters
+
+  if (!sentFromDay && !sentFromMonth && !sentFromYear && !sentToDay && !sentToMonth && !sentToYear && !sentBy) {
+    return null
+  }
+
+  const validation = NotificationsIndexValidator.go(filters)
+
+  if (!validation.error) {
+    return null
+  }
+
+  const toDateError = validation.error.details.find((detail) => {
+    return detail.context.key === 'toFullDate'
+  })
+  const fromDateError = validation.error.details.find((detail) => {
+    return detail.context.key === 'fromFullDate'
+  })
+  const sentByError = validation.error.details.find((detail) => {
+    return detail.context.key === 'sentBy'
+  })
+
+  return {
+    text: 'There was a problem with your filters.',
+    ...(fromDateError && { fromFullDate: { message: fromDateError.message } }),
+    ...(toDateError && { toFullDate: { message: toDateError.message } }),
+    ...(sentByError && { sentBy: { message: sentByError.message } })
+  }
```

Comment:
Could we not just iterate details and add the errors found? This might also solve the problem I spotted: if you add a to date before a from date, it correctly validates there is an error, but this current logic misses displaying that in the summary.

<img width="986" alt="Screenshot 2025-04-13 at 11 41 45" src="https://github.com/user-attachments/assets/c587ee2c-edf1-4a6c-a136-488c50aed1eb" />

Reply:
In fact, validation is going to need a bit more work. Spotted now we are just showing the message `There was a problem with your filters` when we should be showing the specific error messages from Joi. And those errors should when clicked take the user to the problem field.

[GOV.UK Error summary](https://design-system.service.gov.uk/components/error-summary/)

I think we can also avoid most of the prep work in the validator itself, and just validate the `filters` object directly.